File: Cross-platform .NET/Chapter_08/Soundex/soundex.py
#soundex.py - Calls a .NET assembly to calculate a Soundex encoding

import sys
import CLR.Phoneticator as Phoneticator

#Display the program's output
def printresults(name, encoding):
    print "The soundex encoding of", name, "is", encoding

#The program's guts
def main():

    #The script name is always the first argument
    if len(sys.argv) <> 2: 
        print "Please pass in a surname to encode."
        sys.exit(0)
    else:
        surname = sys.argv[1] 
        printresults(surname , Phoneticator.Phoneticator.Soundex(surname))

#Only call the main method if soundex.py has not been imported
if __name__ == '__main__':
    main() 

File: Cross-platform .NET/Chapter_08/Soundex/Soundex.java
//Soundex.java

//Import the .NET Phoneticator namespace
import cli.Phoneticator.*;

public class Soundex
{     	
    //Display the program's output
    private static void printResults(String name, String encoding)
    {
      System.out.println("The soundex encoding of " + name + " is " + encoding);
    }

    public static void main(String args[])       
    {  	
         //The name to encode should be the only argument
         if (args.length != 1)
         {
             System.out.println("Please pass in a surname to encode.");
         }
         else
         {
             String surname = args[0]; 
             printResults(surname, Phoneticator.Soundex(surname));
         }
    }     
}